.changeset/twelve-grapes-dream.md

```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/cloudflare': patch
+---
+
+Fixes an issue where `404.astro` was not used when `routes.strategy` was set to `exclude`
```

Comment:
We might need to update the `changeset` again, once we know why `routeData` behavior changed. Waiting for clarification on that.